docs/proposals/resource-quota.md

```diff
@@ -0,0 +1,453 @@
+
+To facilitate both operational environments, we propose a `ResourceQuota` can
+have a `ResourceAccountingPolicy`.  If set to `Requests`, the quota document will charge
+for quota usage based on the `resources.requests` on a `Kind`.  If set to `Limits`,
+the quota document will charge for quota usage based on `resources.limits` on a `Kind`.
+If not specified, it will default to `Requests` to preserve backward-compatibility.
+If set to `Limits`, a `Kind` must specify an upper limit for each resource tracked
+by quota.  The `ResourceAccountingPolicy` is only pertinent to resources that support
+bursting semantics.
+
+### Filter the set of resources tracked
+
+The set of resources tracked by a `ResourceQuota` can be filtered
+via an optional `LabelSelector` and `FieldSelector`.
+
+A `LabelSelector` matches a resource independent of its kind based on its
+user-defined set of labels.  The label selector supports both equality and
+set-based requirements.  It allows a quota to track resource consumption
+based on local understanding of individual workloads.  For example,
+a quota could match all kinds in a namespace that match the selector
+`tier=cache, environment in (dev,qa)` separately from a quota
+that matched `tier=cache, environment in (prod)`.
+
+A `FieldSelector` groups an `APIVersion` and `Kind`
+with a selector over its fields that supports both equality and
+set-based requirements.  It allows a quota to track objects by
+based on cluster defined attributes.  For example, a quota could match
+all `Pod` objects in a namespace that match the field selector
+`spec.restartPolicy=Always`.  This would enable an administrator
+to quota pods for terminating vs non-terminating workloads differently
+to support fair sharing of a node.  Similarly, if a `Pod` object exposed
+its quality of service class, an administrator could restrict the number
+of `BestEffort` pods via simple selection criteria.  If a quota
+has a `FieldSelector` that restricts the quota to a particular `Kind`,
+the set of `ResourceName` metrics that can be tracked by the
+`ResourceQuota` document are scoped to the set of resources appropriate
+to that `Kind`.  For example, a quota filtered by `Pod` would support
+the resources `memory`, `cpu`, and `pods`.  Any other resource enumerated
+would result in a validation error.
+
+A `ResourceQuota` makes quota decisions based on the object's desired
+state and not the status.  As a result, it's recommended that a
+`FieldSelector` only operate against standard metadata or spec fields.
+
+### Replenishment policies
+
+A `ResourceQuota` could have an optional `ReplenishmentPolicy` that
+defines how a depleted resource is replenished.
+
+Candidate `ReplenishmentPolicy` options include the following:
+* `Normal` - the quota is replenished when the server observes the
+resource is no longer actively consumed (default, current behavior)
+* `Daily` - the quota is replenished at a server-configured time
```

Comment:
So, for Daily, the quota is incremented at admission, but not decremented at deletion.   IT just gets decremented at the replenishment time?

By the way, do we decrement resource quota at deletion or at termination time?  If it is the former, then we really need to fix it, because it will stop you from running long-running jobs (you can't delete pods while a job is still running).